src/app/app-common/include/OpenKneeboard/ImguiInstance.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <OpenKneeboard/CursorEvent.h>
+#include <OpenKneeboard/DXResources.h>
+#include <OpenKneeboard/Events.h>
+#include <OpenKneeboard/ThreadGuard.h>
+
+#include <OpenKneeboard/inttypes.h>
+
+#include <unordered_set>
+
+#include <imgui.h>
+
+namespace OpenKneeboard {
+class KneeboardState;
+
+class ImguiInstance final {
+ public:
+  ImguiInstance(const DXResources&, KneeboardState*);
+  ~ImguiInstance();
+  void Render(ID2D1DeviceContext*, PageID, const D2D1_RECT_F& targetRect);
+
+ private:
+  DXResources mDXR;
+  KneeboardState* mKneeboard;
+  ImGuiContext* mImguiCtx;
+
+  winrt::com_ptr<ID2D1DeviceContext> m2DContext;
+  winrt::com_ptr<ID3D11Texture2D> mTexture;
+  winrt::com_ptr<ID2D1Bitmap1> mBitmap;
```

Comment:
Is this just for testing? I'm not certain it's possible, but I'm hoping the final version can operate directly on the same D2D contexts/textures that are used for the other rendering passes.

Also if this is going to stay around, make sure the pointer members above are initialized to nullptr

Reply:
Doesn't seem so, looks like Dear ImGui expects to render to a 3D context, this was the only way I could find to get it on a 2D Texture and control where it was rendered.

Reply:
Some options:
- The current surfaces are always backed by D3D; perhaps GetTarget().as<ID2D1Bitmap>().GetSurface(), though I’m not sure if that queryinterface is valid
- When setting the D2D context’s target, we could also set the underlying D3D implicit context’s target
- … worst case, could thread through the bitmap/DXGI surface/d3d texture

Reply:
There’s two sources for the usual textures:
- InterprocessRenderer (fully under our control)
- TabPage: partially under our control, but winui3 swapchainpanel is also involved here 

It’s also probably worth noting that these can have different sizes and viewports for the render